### PR TITLE
Stop lr_scheduler.step() being called when not in training mode

### PR DESCRIPTION
**Pull request by jonhare**
Pull Requests to torchbearer are generally pretty good. We only really have one question. If you added, removed or changed something in the main codebase, **did you update CHANGELOG.md???**

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]
LGTM
